app/assets/javascripts/vote.js

```diff
@@ -0,0 +1,49 @@
+var makeAjaxRequestUpPost = function(reviewButtonTo, venueID, reviewId, voteType) {
+  var request = $.ajax({
+    method: 'PATCH',
+    data: { up: voteType },
+    url: '/api/venues/' + venueID + '/reviews/' + reviewId
+  });
+
+  request.done(function(data) {
+    var review = $('#review-' + data.id)[0]
+    review.getElementsByClassName("helpful")[0].innerHTML = "Helpful: " + data.revup_count
+    review.getElementsByClassName("not-helpful")[0].innerHTML = "Not Helpful: " + data.revdown_count
+  });
+};
+
+var makeAjaxRequestDownPost = function(reviewButtonTo, venueID, reviewId, voteType) {
+  var request = $.ajax({
+    method: 'PATCH',
+    data: { down: voteType },
+    url: '/api/venues/' + venueID + '/reviews/' + reviewId
+  });
+
+  request.done(function(data) {
+    var review = $('#review-' + data.id)[0]
+    review.getElementsByClassName("helpful")[0].innerHTML = "Helpful: " + data.revup_count
+    review.getElementsByClassName("not-helpful")[0].innerHTML = "Not Helpful: " + data.revdown_count
+  });
+};
+
+$('#upvote').on('click', function(event) {
+    event.preventDefault();
+    var reviewButton = event.target;
+    var venueId = reviewButton.baseURI.match(/^(.*?)(\d+)(\D*)$/)[2];
+    var reviewButtonTo = $(reviewButton).closest('.button_to');
+    var reviewId = $(reviewButtonTo).attr('action').match(/\/(\d+)\//)[1];
```

Comment:
You can get to the reviewId based on the reviewButtons url. Something along the lines of this

reviewButton.find("button").attr("href").split("/")[5]